server/rating_service/rating_service.py

```diff
@@ -0,0 +1,382 @@
+        )
+
+    async def _get_rating_data(self, summary: GameRatingSummary) -> GameRatingData:
+        ratings = {}
+        for team in summary.teams:
+            for player_id in team.player_ids:
+                ratings[player_id] = await self._get_player_rating(
+                    player_id, summary.rating_type
+                )
+
+        return [
+            TeamRatingData(
+                team.outcome,
+                {player_id: ratings[player_id] for player_id in team.player_ids},
+            )
+            for team in summary.teams
+        ]
+
+    async def _get_player_rating(
+        self, player_id: int, rating_type: RatingType
+    ) -> Rating:
+        if self._rating_type_ids is None:
+            self._logger.warning(
+                "Tried to fetch player data before initializing service."
+            )
+            raise ServiceNotReadyError("RatingService not yet initialized.")
+
+        rating_type_id = self._rating_type_ids.get(rating_type.value)
+        if rating_type_id is None:
+            raise ValueError(f"Unknown rating type {rating_type}.")
+
+        async with self._db.acquire() as conn:
+            sql = select([rating_table.c.mean, rating_table.c.deviation]).where(
+                and_(
+                    rating_table.c.login_id == player_id,
+                    rating_table.c.leaderboard_id == rating_type_id,
+                )
+            )
+
+            result = await conn.execute(sql)
+            row = await result.fetchone()
+
+        if not row:
+            return await self._get_player_legacy_rating(player_id, rating_type)
+
+        return Rating(row[rating_table.c.mean], row[rating_table.c.deviation])
+
+    async def _get_player_legacy_rating(
+        self, player_id: int, rating_type: RatingType
+    ) -> Rating:
+        if rating_type is RatingType.GLOBAL:
+            table = legacy_global_table
+            sql = select([table.c.mean, table.c.deviation, table.c.numGames]).where(
+                table.c.id == player_id
+            )
+        elif rating_type is RatingType.LADDER_1V1:
+            table = legacy_ladder1v1_table
+            sql = select(
+                [table.c.mean, table.c.deviation, table.c.numGames, table.c.winGames]
+            ).where(table.c.id == player_id)
+        else:
+            raise ValueError(f"Unknown rating type {rating_type}.")
+
+        async with self._db.acquire() as conn:
+
+            result = await conn.execute(sql)
+            row = await result.fetchone()
+
+            if not row:
+                new_rating = await self._create_default_rating(
+                    conn, player_id, rating_type
+                )
+                return new_rating
+
+            if rating_type is RatingType.GLOBAL:
+                # The old `global_rating` table does not have a `winGames` column.
+                # This should be a decent approximation though.
+                won_games = row[table.c.numGames] // 2
```

Comment:
We could probably query for this, but it would be pretty annoying. I think we'd have to look at the game_player_stats before/after rating values to see if they went up or down.